pkg/reconciler/labeler/labels.go

```diff
@@ -157,21 +159,31 @@ func deleteLabelForNotListed(ctx context.Context, ns, name string, acc accessor,
 // a nil route name will cause the route label to be deleted, and a non-nil route will cause
 // that route name to be attached to the element.
 func setRouteLabel(acc accessor, elt kmeta.Accessor, routeName *string) error {
-	mergePatch := map[string]interface{}{
-		"metadata": map[string]interface{}{
-			"labels": map[string]interface{}{
-				serving.RouteLabelKey: routeName,
+	return reconciler.RetryUpdateConflicts(func(attempt int) (err error) {
+		// Refetch the respective entity from API on subsequent attempts.
+		if attempt > 0 {
+			elt, err = acc.getApi(elt.GetNamespace(), elt.GetName())
+			if err != nil {
+				return err
+			}
+		}
+
+		mergePatch := map[string]interface{}{
+			"metadata": map[string]interface{}{
+				"labels": map[string]interface{}{
+					serving.RouteLabelKey: routeName,
+				},
+				"resourceVersion": elt.GetResourceVersion(),
```

Comment:
@mattmoor I was wondering why we need this resource version in the patch? Are we using this to prevent a bump in the version and thus an event being sent? Or is it just to ensure consistency?

Reply:
Discussed this with Matt and we agreed that it's not necessary to keep.